specification/recoveryservices/Recoveryservices.Management/models.tsp

```diff
@@ -0,0 +1,1685 @@
+
+  /**
+   * The State of the Resource after the move operation
+   */
+  @visibility(Lifecycle.Read)
+  moveState?: ResourceMoveState;
+
+  /**
+   * Backup storage version
+   */
+  @visibility(Lifecycle.Read)
+  backupStorageVersion?: BackupStorageVersion;
+
+  /**
+   * property to enable or disable resource provider inbound network traffic from public clients
+   */
+  publicNetworkAccess?: PublicNetworkAccess;
+
+  /**
+   * Monitoring Settings of the vault
+   */
+  monitoringSettings?: MonitoringSettings;
+
+  /**
+   * Restore Settings of the vault
+   */
+  restoreSettings?: RestoreSettings;
+
+  /**
+   * The redundancy Settings of a Vault
+   */
+  redundancySettings?: VaultPropertiesRedundancySettings;
+
+  /**
+   * Security Settings of the vault
+   */
+  securitySettings?: SecuritySettings;
+
+  /**
+   * Secure Score of Recovery Services Vault
+   */
+  @visibility(Lifecycle.Read)
+  secureScore?: SecureScoreLevel;
+
+  /**
+   * Security levels of Recovery Services Vault for business continuity and disaster recovery
+   */
+  @visibility(Lifecycle.Read)
+  bcdrSecurityLevel?: BCDRSecurityLevel;
+
+  /**
+   * ResourceGuardOperationRequests on which LAC check will be performed
+   */
+  resourceGuardOperationRequests?: string[];
+}
+
+/**
+ * Details for upgrading vault.
+ */
+model UpgradeDetails {
+  /**
+   * ID of the vault upgrade operation.
+   */
+  @visibility(Lifecycle.Read)
+  operationId?: string;
+
+  /**
+   * UTC time at which the upgrade operation has started.
+   */
+  @visibility(Lifecycle.Read)
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  startTimeUtc?: utcDateTime;
+
+  /**
+   * UTC time at which the upgrade operation status was last updated.
+   */
+  @visibility(Lifecycle.Read)
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  lastUpdatedTimeUtc?: utcDateTime;
+
+  /**
+   * UTC time at which the upgrade operation has ended.
+   */
+  @visibility(Lifecycle.Read)
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  endTimeUtc?: utcDateTime;
+
+  /**
+   * Status of the vault upgrade operation.
+   */
+  @visibility(Lifecycle.Read)
+  status?: VaultUpgradeState;
+
+  /**
+   * Message to the user containing information about the upgrade operation.
+   */
+  @visibility(Lifecycle.Read)
+  message?: string;
+
+  /**
+   * The way the vault upgrade was triggered.
+   */
+  @visibility(Lifecycle.Read)
+  triggerType?: TriggerType;
+
+  /**
+   * Resource ID of the upgraded vault.
+   */
+  @visibility(Lifecycle.Read)
+  upgradedResourceId?: string;
+
+  /**
+   * Resource ID of the vault before the upgrade.
+   */
+  @visibility(Lifecycle.Read)
+  previousResourceId?: string;
+}
+
+/**
+ * Information to be stored in Vault properties as an element of privateEndpointConnections List.
+ */
+model PrivateEndpointConnectionVaultProperties {
```

Comment:
>PrivateEndpointConnectionVaultProperties [](http://example.com/codeflow?start=6&length=40)

why not just PrivateEndpointConnectionProperties?

Reply:
We cannot. They are not referring to the common types. We cannot change it into the common type one by ourselves.